ledger/complete/mtrie/trie/trie.go

```diff
@@ -250,14 +270,20 @@ func (parentTrie *MTrie) update(nodeHeight int, parentNode *node.Node,
 
 	// recurse over each branch
 	var lChild, rChild *node.Node
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+	if len(lpayloads) == 0 || len(rpayloads) == 0 {
```

Comment:
Good idea! 
I think I should generalize this check to `if len(lpayloads) < threshold || len(rpayloads) < threshold`, and make tests on my machine for a optimum `threshold` value .